core/templates/dev/head/pages/exploration-editor-page/services/user-exploration-permissions.service.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview A service that fetches and stores the permissions
+ * of a user for a particular exploration.
+ */
+require('domain/utilities/UrlInterpolationService.ts');
+require('services/ContextService.ts');
+require('services/contextual/UrlService.ts');
+
+require(
+  'pages/exploration-editor-page/exploration-editor-page.constants.ajs.ts');
+
+angular.module('oppia').factory('UserExplorationPermissionsService', [
+  '$http', '$q', 'ContextService', 'UrlInterpolationService', 'UrlService',
+  'EXPLORATION_RIGHTS_URL',
+  function(
+      $http, $q, ContextService, UrlInterpolationService, UrlService,
+      EXPLORATION_RIGHTS_URL) {
+    var permissionsPromise = null;
+    var pathname = UrlService.getPathname();
+
+    return {
+      getPermissionsAsync: function() {
+        // TODO(#7221): This can be removed after exploration data
+        // service works correctly if called from different contexts.
+        if (pathname.includes('collection_editor')) {
```

Comment:
As we know the cause of this issue should we move this check around the caller (Want to avid such cases to happen in other places. )

Reply:
I think I have done something similar in #7385. So this issue should be fixed when that goes in.

Reply:
I mean, if you know which method is calling `getPermissionsAsync` from the collection_editor page then prevent that method to call this and add a top-level comment over there to resolve this issue. 

Let's not make service to handel this "unexpected" case as other directives/services are using this method as well and if we are going to handel the check over here then we are allowing anyone to create another such unexpected behavior without knowing.

Reply:
It is being called from the exploration data service.
After the unification of the autosave URLs, this doesn't happen again.